src/sources/aws_s3/sqs.rs

```diff
@@ -122,6 +122,19 @@ pub(super) struct Config {
     #[configurable(metadata(docs::examples = 5))]
     pub(super) client_concurrency: Option<NonZeroUsize>,
 
+    /// Maximum number of messages to poll from sqs in a batch
+    ///
+    /// Defaults to 10
+    ///
+    /// Should be set to smaller numbers when the files are larger to help prevent large ingestions
+    /// from causing other files to exceed the visibility_timeout. Valid values are 1 - 10
```

Comment:
```suggestion
    /// Maximum number of messages to poll from sqs in a batch (default 10). Valid values are 1-10.
    ///
    /// Should be set to smaller numbers when the files are larger to help prevent large ingestions
    /// from causing other files to exceed the `visibility_timeout`.
```

Reply:
If you set it to an invalid value, the sqs fetch fails:

```
2024-04-08T20:24:42.827559Z ERROR source{component_kind="source" component_id=sqs_test component_type=aws_s3}: vector::internal_events::aws_sqs: Failed to fetch SQS events. error=service error error_code="failed_fetching_sqs_events" error_type="request_failed" stage="receiving" internal_log_rate_limit=true
2024-04-08T20:24:42.827649Z ERROR source{component_kind="source" component_id=sqs_test component_type=aws_s3}: vector::internal_events::aws_sqs: Internal log [Failed to fetch SQS events.] is being suppressed to avoid flooding.
```

While I like validation and more informative error messages, I couldn't find an example of another config value like this being validated, and I don't actually speak rust so I simply don't know how/where I would do it.

Reply:
It'd look something like:

- Add a new enum variant to https://github.com/vectordotdev/vector/blob/9080d053edff3aa520be857f9619eac5345b8654/src/sources/aws_s3/sqs.rs#L143-L150
- Check in https://github.com/vectordotdev/vector/blob/9080d053edff3aa520be857f9619eac5345b8654/src/sources/aws_s3/sqs.rs#L222-L230 if the number is in-bounds and return the newly created error if not

I realize it might be a bit of a lift if you are unfamiliar with Rust though. If you can't get to it I can try to get to it soon. I think this that this would be a common error for users to run into so would prefer to handle it upfront if we can.

Reply:
Thank you. I think I got this, but you can review and let me know if I did something wrong.

Generates error messages on `vector validate` of the format:
```
Component errors
----------------
x Source "sqs_test": Invalid value for max_number_of_messages 0
```